File: AppCore/src/main/java/com/rkarp/appcore/model/ChangelogEntry.java
package com.rkarp.appcore.model;

import javafx.beans.property.SimpleStringProperty;

/**
 * Created by rickie on 3/12/2016.
 */
public class ChangelogEntry {

    private final SimpleStringProperty version;
    private final SimpleStringProperty date;
    private final SimpleStringProperty desc;

    public ChangelogEntry(String version, String date, String desc) {
        this.version = new SimpleStringProperty(version);
        this.date = new SimpleStringProperty(date);
        this.desc = new SimpleStringProperty(desc);
    }

    public String getVersion() {
        return version.get();
    }
    public String getDate() {
        return date.get();
    }
    public String getDesc() {
        return desc.get();
    }

}

File: projects/JavaFXApps/core/src/main/java/net/rickiekarp/core/model/ChangelogEntry.kt
package net.rickiekarp.core.model

import javafx.beans.property.SimpleStringProperty

/**
 * Created by rickie on 3/12/2016.
 */
class ChangelogEntry(version: String, date: String, desc: String) {
    private val version: SimpleStringProperty = SimpleStringProperty(version)
    private val date: SimpleStringProperty = SimpleStringProperty(date)
    private val desc: SimpleStringProperty = SimpleStringProperty(desc)

    fun getVersion(): String {
        return version.get()
    }

    fun getDate(): String {
        return date.get()
    }

    fun getDesc(): String {
        return desc.get()
    }

}
